src/main/java/dev/blaauwendraad/masker/json/MaskingState.java

```diff
@@ -104,23 +104,31 @@ public void backtrackCurrentJsonPath() {
         currentJsonPath.pop();
     }
 
-    /**
-     * Checks if the last segment of the current jsonpath is an array index.
-     */
-    public boolean isInArray() {
-        return !currentJsonPath.isEmpty() && currentJsonPath.peek().offset == -1;
-    }
-
     /**
      * Increments an array index in the last segment of the current jsonpath.
      * Throws {@link java.lang.IllegalStateException} if the last segment is not an array index.
      */
     public void incrementCurrentJsonPathArrayIndex() {
         SegmentReference lastSegment = currentJsonPath.peek();
-        assert isInArray() && lastSegment != null;
+        assert isInArraySegment() && lastSegment != null;
```

Comment:
I think `assert`'s in production code is usually considered bad practice. 

Wouldn't it be better to just do `if (condition)`  with an IllegalStateException with a clear message what is going wrong? 

This might impact performance actually (we saw something like that in the past) so you might want to check benchmarks before/after

Reply:
Also: consider if we need this check in the first place: is it some internal code that calls this and can we make sure this never happens? If so, we might consider removing the check altogether

Reply:
> is it some internal code that calls this and can we make sure this never happens

Exactly. I will remove it as If we are finished with adding more logic on top of the current masking loop. I am curious though about what would benchmarks say about "assertions" against "exceptions" performance.